src/relayserver/ReputationManager.ts

```diff
@@ -82,27 +96,81 @@ export class ReputationManager {
     return PaymasterStatus.GOOD
   }
 
+  createNewEntry (paymaster: string): ReputationEntry {
+    const newEntry = {
+      paymaster: paymaster.toLowerCase(),
+      reputation: this.config.initialReputation,
+      lastAcceptedRelayRequestTs: 0,
+      abuseStartedBlock: 0,
+      changes: []
+    }
+    this.localReputationEntries.set(paymaster.toLowerCase(), newEntry)
+    return newEntry
+  }
+
   async onRelayRequestAccepted (paymaster: Address): Promise<void> {
+    /*
     await this.reputationStoreManager.updateLastAcceptedTimestamp(paymaster)
+     */
+    const lastAcceptedRelayRequestTs = Date.now()
+    const entry = this.localReputationEntries.get(paymaster.toLowerCase()) ??
+      this.createNewEntry(paymaster)
+    entry.lastAcceptedRelayRequestTs = lastAcceptedRelayRequestTs
+    this.logger.debug(`Paymaster ${paymaster} was last accepted at ${lastAcceptedRelayRequestTs}`)
   }
 
-  async updatePaymasterStatus (paymaster: Address, transactionSuccess: boolean): Promise<void> {
+  async updatePaymasterStatus (paymaster: Address, transactionSuccess: boolean, eventBlockNumber: number): Promise<void> {
     const change = transactionSuccess ? 1 : -1
+    /*
     const entry =
       await this.reputationStoreManager.getEntry(paymaster)
+     */
+    const entry = this.localReputationEntries.get(paymaster.toLowerCase()) ??
+      this.createNewEntry(paymaster)
     if (entry == null) {
       throw new Error(`Could not query reputation for paymaster: ${paymaster}`)
     }
     if (entry.reputation + change > this.config.maximumReputation) {
       return
     }
     const changeInAbuseWindow = entry.changes
-      .filter(it => Date.now() - it.timestamp < this.config.abuseTimeWindowMs)
+      .filter(it => eventBlockNumber - it.blockNumber < this.config.abuseTimeWindowBlocks)
       .reduce((previousValue: number, currentValue: ReputationChange) => previousValue + currentValue.change, 0)
     if (-changeInAbuseWindow >= this.config.abuseReputationChange) {
+      /*
       await this.reputationStoreManager.setAbuseFlag(paymaster)
+       */
+      entry.abuseStartedBlock = eventBlockNumber
     }
-    const oldChangesExpirationTs = Date.now() - this.config.abuseTimeWindowMs
+    const oldChangesExpirationBlock = eventBlockNumber - this.config.abuseTimeWindowBlocks
+    /*
     await this.reputationStoreManager.updatePaymasterReputation(paymaster, change, oldChangesExpirationTs)
+     */
+    this.updatePaymasterReputation(paymaster, change, oldChangesExpirationBlock, eventBlockNumber)
+  }
+
+  updatePaymasterReputation (
```

Comment:
why this method is duplicated?

Reply:
One is in the Storage Manager. I do not remove the persistent storage code yet as I believe we will reenable it quite soon. 
The other is "re-implementation" with a `Map` instance inside the Reputation Manager.